### ISSUE TITLE
Code completion does not work for fields of the parent class when using @:structInit metadata.

### ISSUE THREAD

**Issue by RedHairHead** [HaxeFoundation/haxe#9145]
```haxe
@:publicFields
@:structInit
class A {
    var x:Int;
}

@:publicFields
@:structInit
class B extends A {
    var y:Int;
}

class StructTest {
    public function new() {
        var b:B = {};
    }
}
```
![structInit](https://user-images.githubusercontent.com/50680881/74587043-c4c8c180-4ffe-11ea-845c-5241f24c80ad.png)

**Comment by Simn** [comment]
Wait that's allowed...?

**Comment by RedHairHead** [comment]
As I understand it, this is a new feature. I learned the other day about her in a discord сhannel.

**Comment by Simn** [comment]
I know about `@:structInit`, I just didn't remember that it supported inheritance like that.

**Comment by RedHairHead** [comment]
Well, I tried to do this and it works, despite the lack of code completion.

**Comment by RealyUniqueName** [comment]
@Simn https://github.com/HaxeFoundation/haxe/blob/e849861/src/typing/typeloadFields.ml#L132

**Comment by RedHairHead** [comment]
I also add that code completion for `@:structInit` shows fields that cannot be set, which would also be nice to fix.

```haxe
@:publicFields
@:structInit
class A {
	var x:Int;
	function foo():Void {}
}

class Main {
	static function main() {
		var a:A = {};
	}
}
```
![structInit](https://user-images.githubusercontent.com/50680881/74654370-3473b380-519b-11ea-9bf7-b7e74f1485ca.png)

**Comment by nadako** [comment]
added the partner label because we encoured this too and it would be nice to fix

**Comment by RealyUniqueName** [comment]
Fixed by #9263
